controllers/mdutil/util.go

```diff
@@ -693,22 +693,28 @@ func CloneSelectorAndAddLabel(selector *metav1.LabelSelector, labelKey, labelVal
 // DeepHashObject writes specified object to hash using the spew library
 // which follows pointers and prints actual values of the nested objects
 // ensuring the hash does not change when a pointer changes.
-func DeepHashObject(hasher hash.Hash, objectToWrite interface{}) {
+func DeepHashObject(hasher hash.Hash, objectToWrite interface{}) error {
```

Comment:
This breaks API compatibility, we should deprecate this method and add a new one instead that returns an error, for example it'd also be great to rename it as `SpewHashObject` to signal that we're using the Spew library to hash objects.

/hold